services/ske/wait_test.go

```diff
@@ -179,13 +200,14 @@ func TestCreateOrUpdateClusterWaitHandler(t *testing.T) {
 			name := "cluster"
 
 			apiClient := &apiClientClusterMocked{
-				getFails:      tt.getFails,
-				name:          name,
-				resourceState: tt.resourceState,
+				getFails:             tt.getFails,
+				name:                 name,
+				resourceState:        tt.resourceState,
+				invalidArgusInstance: tt.invalidArgusInstance,
 			}
 			var wantRes *ClusterResponse
 			rs := ClusterStatusState(tt.resourceState)
-			if !tt.getFails {
+			if !tt.getFails && tt.resourceState != stateUnhealthy {
```

Comment:
Doesn't this extra condition make the test not check the hander's response when `tt.invalidArgusInstance=true`? In that case you'd want to check if `gotRes` matches what you'd expect